[WIP] README (#2)
HubRise/tiny-tablet: README.md

<a href="https://www.hubrise.com/"><img src="./screenshots/hubrise-logo.png" align="left" height="48"></a>
<br><br><br>

> # HubRise {{app_name}} Template

Reviewer: "EasyTablet", "TinyTablet", "HelloWorld", "TinyOrder", "MiniOrder"... Any suggestions?

Author: I'd go for TinyTablet!

{{app_name}} is an app that prints new orders received by your HubRise location, with just a few lines of code.

> You can fork this repository and make changes to the code to create your own app. 

Reviewer: The goal is more to showcase what can be achieved in 250 lines of code (HTML included!) than serving as a skeleton for a real app.

## Try the app

First, you need to [create a HubRise account](https://manager.hubrise.com/signup). It's free and it only takes a minute. 

Then, simply [open the app](http://philatist-fish-74534.netlify.com) in your browser. 

You will be asked to connect the app to your HubRise account. 
Granting permissions is a necessary step in every OAuth2.0 authentication process, and requires the user to manually **Allow** the app access. 

![OAuth page granting access to the app](./screenshots/oauth-page.png)

You will then see a single-page app that displays the list of the latest orders received by your HubRise location. 
Each line in the logs displays the time, id, status, and total price of the order.

> ![Single-page app with the orders from HubRise](./screenshots/logs-page.png)

Reviewer: Should we include instructions on how to inject orders? Maybe we could create another client for this single purpose, and share the id and secret in this README to save the user from doing it.

The user would just need to:
- Authorize the new client on his location by opening a predefined URL.
- Copy the authorization code in a predefined curl request to retrieve a token.
- Copy the token in a predefined curl request to create an order.

Author: I think that adding all that information in the README could steal away the attention from the purpose of this app.
I see two options here:
1. We can point the reader to the Quick start guide, where they can find pre-built Postman requests and learn how to inject orders
2. We could create a TinyOrder app, similar to this, where the user can send a predefined order to the app (maybe just changing the customer's name). So they would open the two pages, send the order from one and receive it in the other.

Reviewer: I was thinking initially of redirecting the user to the Quick Start, but that will lead him to take extra steps, like creating a client. We want him to see orders in TinyTablet as fast as possible. So maybe we can point to the Quick Start, and add a TLDR in the form of a few simple steps that he can copy-paste.

Creating a reciprocal app to shoot orders is an interesting idea. A concern however is that it will keep the user (likely a developer) away from the API, that we ultimately want him to get interested in.

How much do you think you could shorten the steps to setup another connection and shoot an order from curl?

Author: I've added a short paragraph under an optional section. I hope it's short enough, I think we cannot do much less than that.

Clicking on one of the logs reveals additional information about the order, such as the customer's details and the purchased products. 

![Modal with additional information about the order](./screenshots/single-order-modal.png)

> ## What this app is NOT

Author: @antmonnier In the issue, you mentioned that the user might want to add a DB. What would be the purpose of that in this case?

Reviewer: Probably if the user was to extend this app, he would feel the need for a DB at some point. But maybe not even worth mentioning in this readme.

Author: OK fine!

This app is not production ready! 
Consider it some boilerplate code that you can buil upon to create your own app.
It is also useful to show some of the things you should _not_ do in your app! 

For example, in the first line of the `js/app.js` file, you will find the client secret. 
This is bad coding practice. 

Also, the user interface is too crude to be used in a real case scenario. 

> ## Prerequisites

Author: @antmonnier Any particular prerequisites we need here?

Reviewer: Nope

...

> ## Running {{app_name}} locally

Author: @antmonnier I guess someone might want to do this. Is it enough to un-comment the first lines in js/app.js to obtain this? I suspect not...

Reviewer: Good idea!

Assuming the user has nodeJS installed, he can use `serve`, a very lightweight JS server for static files.

Installation from the console (note: yarn is a modern & faster alternative to node's default package manager npm):
`npm install -g serve` (or equivalently `yarn global add serve` if you are using yarn)
(this command might require root permissions).

At the root of the project, start the server by typing:
`serve`

And open the browser on the URL shown by the command, typically `http://localhost:5000`

Et voilà!

Author: It works! Witchcraft! 
The only thing I had to change was to add `sudo` before `npm`. I guess it does not hurt to add that.

To run {{app_name}} locally, follow these steps:

Linux and macOS:
```
<run_command>
```

Windows:
```
<run_command>
```

## Additional resources

> The HubRise documentation provides you additional information to get you started and make this app production ready. 

Reviewer: ```suggestion
The HubRise documentation provides you additional information to connect your app to HubRise.
```

To learn more about the HubRise platform, see our [User guide](https://www.hubrise.com/docs/).

To learn how to connect your solution to HubRise, try our [Quick start guide](https://www.hubrise.com/developers/quick-start).

> For more information about the HubRise API, see our [API reference](https://www.hubrise.com/developers/api/general-concepts).

Reviewer: ```suggestion
The HubRise documentation provides you additional information to connect your app to HubRise.

Try our [Quick start guide](https://www.hubrise.com/developers/quick-start) to get you started.

For more information about the HubRise API, see our [API reference](https://www.hubrise.com/developers/api/general-concepts).
```
